app/javascript/src/Policies/components/CustomPolicy.jsx

```diff
@@ -46,17 +46,30 @@ function CSRFToken ({win = window}: {win?: any}) {
 
 function CustomPolicyForm ({policy, onChange}: {policy: Policy, onChange: OnChange}) {
   const isNewPolicy = !(policy.id && policy.id !== 0)
-  const action = (isNewPolicy) ? '/p/admin/registry/policies/' : `/p/admin/registry/policies/${policy.id}`
+  const initialState = () => {
+    return (isNewPolicy)
+      ? { action: '/p/admin/registry/policies/', method: 'post', submitText: 'Create Custom Policy' }
+      : { action: `/p/admin/registry/policies/${policy.id}`, method: 'put', submitText: 'Update Policy Schema' }
+  }
+  const [state, setState] = useState(initialState())
+
   return (
-    <form action={action} method="post">
+    <form action={state.action} method="post">
       <label>
         Directory:
         <input type="text" name="directory" value={policy.directory} onChange={onChange} disabled={!isNewPolicy} />
       </label>
       <input type="hidden" name="id" value={policy.id} disabled={true} />
       <input name="schema" type="hidden" value={JSON.stringify(policy.schema)} />
-      {(!isNewPolicy) ? <input name="_method" type="hidden" value='put' /> : ''}
-      <input type="submit" />
+      {(!isNewPolicy) &&
+        (
+          <div>
+            <input name="_method" type="hidden" value={state.method} />
+            <input type="submit" value="Delete Policy" onClick={() => setState({...state, ...{method: 'delete'}})} />
```

Comment:
This looks a bit uncontrolled. The idea is that `method` will be `'delete'` by the time submit happens, isn't it?
Would it be possible to give names to our submits, for instance `<input name="action_delete" ... />` so that we don't rely on variables? It would render the necessary submit buttons: `create`, `update` and `delete` and the form would be submitted with only one of those.
This would also simplify the current implementation, since now we need an extra prop `_method` plus 2 submit buttons.
WDYT?

Reply:
(Summing up.. I think the hook is unnecessary)

Reply:
I did it that way since we are letting Rails managing the submission , validation and routing of the Custom Policies. And what it's needed to make it work in Rails, it's that hidden input.

Reply:
OK I see the limitations here. However I still think this is more complex that it should for a React component.
I know this won't sound ideal but since the only relevant field is `_method = 'delete'` when deleting... what about rendering a different form instead of simply a div? I think this way it would be much more simpler, I made a little example: https://codepen.io/anon/pen/jJoRLK

Reply:
That'd be much more complex and will still need to have some duplicated policy info in the second form, which rails is expecting as params. The hidden input _method enhance the default form feature of only get and post, adding put and delete.

Reply:
OK then